pkg/cmd/util/clientcmd/factory.go

```diff
@@ -714,6 +731,10 @@ func (f *Factory) PodForResource(resource string, timeout time.Duration) (string
 	case api.Resource("pods"):
 		return name, nil
 	case api.Resource("replicationcontrollers"):
```

Comment:
Right. I must use `f.Client()` instead of `f.Clients()` so it only takes the kube client. Fixed.